### PR TITLE
elasticsearch-jmeter: Modified to include prereqs directory

### PR DESCRIPTION
**Pull request by PraveenAnil**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: elasticsearch-jmeter/prereqs/prereq.azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,793 @@
+﻿{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+
+    },
+    "variables": {
```
Comment (bmoore-msft) [inline elasticsearch-jmeter/prereqs/prereq.azuredeploy.json]: many of these variables are only used once - let's just use literal values in those cases... the variables just make the template harder to read.

File: elasticsearch-jmeter/prereqs/prereq.azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,793 @@
...
+                    "publisher": "Microsoft.Azure.Extensions",
+                    "type": "CustomScript",
+                    "autoUpgradeMinorVersion": true,
+                    "typeHandlerVersion": "2.0",
+                    "settings": {
+                        "fileUris": "[variables('ubuntuScripts')]",
+                        "commandToExecute": "[concat('bash elasticsearch-ubuntu-install.sh -wn ', variables('esSettings').marvelClusterName, ' -v ', variables('esSettings').version, ' -d ', variables('esSettings').marvelHosts, ' -k ')]"
+                    }
+                },
+                "master": {
```
Comment (bmoore-msft) [inline elasticsearch-jmeter/prereqs/prereq.azuredeploy.json]: these vars seem to differ only in the cmd to execute, let's reduce the var to just the part that varies.

File: elasticsearch-jmeter/prereqs/prereq.azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,793 @@
...
+                    "publisher": "Microsoft.Compute",
+                    "type": "CustomScriptExtension",
+                    "typeHandlerVersion": "1.4",
+                    "autoUpgradeMinorVersion": true,
+                    "settings": {
+                        "fileUris": "[variables('windowsScripts')]",
+                        "commandToExecute": "[concat('powershell.exe -File elasticsearch-windows-install.ps1 -elasticSearchVersion ', variables('esSettings').version, ' -elasticClusterName ', variables('esSettings').marvelClusterName, ' -discoveryEndpoints ', variables('esSettings').marvelHosts, ' -marvelOnlyNode -storageKey ')]"
+                    }
+                },
+                "master": {
```
Comment (bmoore-msft) [inline elasticsearch-jmeter/prereqs/prereq.azuredeploy.json]: simplify, same as the other extension...

File: elasticsearch-jmeter/prereqs/prereq.azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,793 @@
...
+                    "autoUpgradeMinorVersion": true,
+                    "settings": {
+                        "fileUris": "[variables('windowsScripts')]",
+                        "commandToExecute": "[concat('powershell.exe -File elasticsearch-windows-install.ps1 -elasticSearchVersion ', variables('esSettings').version, ' -elasticClusterName ', variables('esSettings').clusterName, ' -discoveryEndpoints ', variables('esSettings').discoveryHosts, variables('cloudAzureParamValue'), variables('marvelParamValue'), variables('marvelExportParamValue'), ' -dataOnlyNode -storageKey ')]"
+                    }
+                }
+            }
+        },
+        "osSettings": "[variables(concat(variables('OS'), 'Settings'))]",
+        "dataSkuSettings": {
```
Comment (bmoore-msft) [inline elasticsearch-jmeter/prereqs/prereq.azuredeploy.json]: can we use a property copy loop for these?

File: elasticsearch-jmeter/prereqs/prereq.azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,793 @@
...
+                    },
+                    "namespace": {
+                        "value": "[concat(variables('esClusterName'), '-kibana')]"
+                    }
+                }
+            }
+        }
+    ],
+    "outputs": {
+        "existingvnetname": {
```
Comment (bmoore-msft) [inline elasticsearch-jmeter/prereqs/prereq.azuredeploy.json]: camelCase

**Comment by PraveenAnil** [comment]
Thanks for the feedback, I will make the required changes.

**Comment by PraveenAnil** [comment]
@bmoore-msft I have made the suggested changes.

File: elasticsearch-jmeter/prereqs/prereq.azuredeploy.json [hunk 6]
```diff
@@ -1,201 +1,56 @@
...
-            "sshPublicKey": {
-                "disablePasswordAuthentication": "true",
-                "ssh": {
-                    "publicKeys": [{
-                        "path": "[concat('/home/', variables('adminUsername'), '/.ssh/authorized_keys')]",
-                        "keyData": "[variables('sshPublicKey')]"
-                    }]
-                }
-            }
+            "password": "adminPassword1!",
```
Comment (bmoore-msft) [inline elasticsearch-jmeter/prereqs/prereq.azuredeploy.json]: Can we use GEN-PASSWORD here?  Also, can you double check that we aren't hard-coding any other passwords in this sample...

Comment (PraveenAnil) [reply elasticsearch-jmeter/prereqs/prereq.azuredeploy.json]: Sure. I have modified the template and made sure no passwords are hard coded. Thanks.

**Pull request merged**